src/app-logic/published-profiles-store.js

```diff
@@ -0,0 +1,128 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
I left this review to the end of the week, and don't have time to finish it. I'll get right on it on Monday though. This is as far as I've gotten. I realize that I need more focus time on this next part.